### ISSUE TITLE
Segmentation fault

### ISSUE THREAD

**Issue by ko1nksm** [groovenauts/blocks-concurrent-subscriber#19]
Segmentation fault has occurred.
I tested it with the following script.

test-script.sh

```shell
#!/bin/sh
set -e
i=0
while true; do
  i=$((i+1))
  printf "$i "
  ./blocks-concurrent-subscriber_darwin_amd64 --version
done
```

```console
$ time ./test-script.sh
1 blocks-concurrent-subscriber version 0.2.3-alpha3
2 blocks-concurrent-subscriber version 0.2.3-alpha3
3 blocks-concurrent-subscriber version 0.2.3-alpha3

(snip)

2593 blocks-concurrent-subscriber version 0.2.3-alpha3
2594 blocks-concurrent-subscriber version 0.2.3-alpha3
2595 blocks-concurrent-subscriber version 0.2.3-alpha3
2596 blocks-concurrent-subscriber version 0.2.3-alpha3
2597 blocks-concurrent-subscriber version 0.2.3-alpha3
2598 blocks-concurrent-subscriber version 0.2.3-alpha3
2599 ./test-script.sh: line 10: 16770 Segmentation fault: 11  ./blocks-concurrent-subscriber_darwin_amd64 --version
./test-script.sh 15.68s user 18.23s system 78% cpu 43.392 total
```

**Comment by akm** [comment]
@ko1nksm Thank you for your report.

I tried your script with `0.2.3-alpha3` compiled by go-1.8.3, but any error doesn't occur.
So I'll fix `.travis.yml` to use `go-1.8.3`.

```
$ ./test-script.sh
1 blocks-concurrent-subscriber version 0.2.3-alpha3
2 blocks-concurrent-subscriber version 0.2.3-alpha3
3 blocks-concurrent-subscriber version 0.2.3-alpha3
(snip)
10246 blocks-concurrent-subscriber version 0.2.3-alpha3
10247 blocks-concurrent-subscriber version 0.2.3-alpha3
10248 blocks-concurrent-subscriber version 0.2.3-alpha3
10249 blocks-concurrent-subscriber version 0.2.3-alpha3
10250 ^C
```

**Comment by ko1nksm** [comment]
Issue was fixed in 0.2.3. thanks.
